screens/auth/SignUpScreen.js

```diff
@@ -92,193 +102,142 @@ export default class SignUp extends React.Component {
-    }
-    // In case we want to do name checking using validate.js
-    // const nameError = validate('name', this.state.name)
+    return signUpPermission;
+  };
 
-    let formattedPhoneNumber = this.state.phoneNumber;
-    // eslint-disable-next-line prettier/prettier
-    formattedPhoneNumber = `(${formattedPhoneNumber.slice(
-      0,
-      3
-    )}) ${formattedPhoneNumber.slice(3, 6)}-${formattedPhoneNumber.slice(
-      6,
-      10
-    )}`;
-    // Update sign up permission based on whether there are any errors left
-    let signUpPermission = false;
-    if (!phoneNumberError && !passwordError && !nameError) {
-      signUpPermission = true;
+  // Add customer to Airtable, creating a push token record first.
+  addCustomer = async () => {
+    const { token, values } = this.state;
+    const name = values[signUpFields.NAME];
+    const phoneNumber = values[signUpFields.PHONENUM];
+    const password = values[signUpFields.PASSWORD];
+    try {
+      const pushTokenId = await createPushTokens({ token });
+      const customerId = await createCustomers({
+        name,
+        phoneNumber,
+        password,
+        points: 0,
+        pushTokenIds: [pushTokenId],
+      });
+      return customerId;
+    } catch (err) {
+      console.error('[SignUpScreen] (addCustomer) Airtable:', err);
     }
-
-    // Have to await this or else Airtable call may happen first
-    await this.setState({
-      nameError,
-      phoneNumberError,
-      passwordError,
-      signUpPermission,
-    });
-    return formattedPhoneNumber;
   };
 
-  // Handling submission. This function runs the validation functions
-  // as well as the duplicate checking. If there are no errors on the
-  // validation or duplicate side, then an Airtable record is created.
-  async handleSubmit() {
-    const formattedPhoneNumber = this.updateErrors();
-    // If we don't have any bugs already with form validation,
-    // we'll check for duplicates here in the Airtable.
-    if (!this.state.phoneNumberError) {
-      const that = this;
-      await checkForDuplicateCustomer(formattedPhoneNumber).then(
-        async resolvedValue => {
-          if (resolvedValue) {
-            // Again, must await this
-            await that.setState({
-              phoneNumberError: 'Phone number in use already.',
-            });
-          }
+  // Handle form submission. Validate fields first, then check duplicates.
+  // If there are no errors, add customer to Airtable.
+  handleSubmit = async () => {
+    const { signUpPermission } = this.state;
+    if (signUpPermission) {
+      try {
+        // Check for duplicates first
+        const formattedPhoneNumber = formatPhoneNumber(
+          this.state.values[signUpFields.PHONENUM]
+        );
+        const duplicateCustomers = await getCustomersByPhoneNumber(
+          formattedPhoneNumber
+        );
+        if (duplicateCustomers.length !== 0) {
+          console.log('Duplicate customer');
```

Comment:
It still let me create duplicate customers if I just spam click the button. I could definitely see this being an issue since it takes a while to see any response and there's no loading screen...
![image](https://user-images.githubusercontent.com/21160510/78269726-d424a100-74be-11ea-890b-e73c0972d2b1.png)

Reply:
Fixed this by adding a `processing` variable in state.